front/app/components/admin/PostManager/components/FilterSidebar/FilterSidebarPhasesItem.tsx

```diff
@@ -1,72 +1,53 @@
 import React from 'react';
 import { IPhaseData, canContainIdeas } from 'services/phases';
 import { Menu, Label } from 'semantic-ui-react';
-import { DropTarget } from 'react-dnd-cjs';
+import { useDrop } from 'react-dnd';
 import T from 'components/T';
 
 interface Props {
   phase: IPhaseData;
   active: boolean;
   onClick: any;
-  isOver: boolean;
-  canDrop: boolean;
   phaseNumber: number;
-  connectDropTarget: any;
 }
 
-class FilterSidebarPhasesItem extends React.PureComponent<Props> {
-  render() {
-    const {
-      phase,
-      active,
-      onClick,
-      connectDropTarget,
-      isOver,
-      canDrop,
-      phaseNumber,
-    } = this.props;
-    const disabled = !canContainIdeas(phase);
-    return connectDropTarget(
-      <div>
-        <Menu.Item
-          active={active || (isOver && canDrop)}
-          onClick={onClick}
-          disabled={disabled}
-        >
-          <Label
-            circular={true}
-            basic={true}
-            color={disabled ? 'grey' : 'teal'}
-          >
-            {phaseNumber}
-          </Label>
-          <T value={phase.attributes.title_multiloc} />
-        </Menu.Item>
-      </div>
-    );
-  }
-}
-
-const phaseTarget = {
-  drop(props: Props) {
-    return {
+const FilterSidebarPhasesItem = ({
+  phase,
+  active,
+  onClick,
+  phaseNumber,
+}: Props) => {
+  const [{ canDrop, isOver }, drop] = useDrop({
+    accept: 'IDEA',
+    drop: () => ({
       type: 'phase',
-      id: props.phase.id,
-    };
-  },
-  canDrop(props: Props) {
-    return canContainIdeas(props.phase);
-  },
-};
+      id: phase.id,
+    }),
+    canDrop() {
+      return canContainIdeas(phase);
+    },
+    collect: (monitor) => ({
+      isOver: monitor.isOver(),
+      canDrop: monitor.canDrop(),
+    }),
+  });
```

Comment:
Replaces the old HoC and we're using the maintained (and typed!) `react-dnd` instead of legacy` react-dnd-cjs`.